crates/storage-rocksdb/src/codec.rs

```diff
@@ -123,26 +122,26 @@ impl<'a> Codec for &'a [u8] {
     }
 }
 
-pub struct VersionedValue<M, S>(pub M, PhantomData<S>);
+pub struct StorageSerdeValue<V>(pub V);
 
-impl<M, S> VersionedValue<M, S> {
-    pub fn new(message: M) -> Self {
-        Self(message, PhantomData::default())
+impl<V> StorageSerdeValue<V> {
+    pub fn into_inner(self) -> V {
+        self.0
     }
 }
 
-impl<M, S> Codec for VersionedValue<M, S>
+impl<V> Codec for StorageSerdeValue<V>
 where
-    S: VersionedSerializer<Value = M>,
+    V: StorageSerde,
 {
```

Comment:
We still need `Codec` for the keys, I believe. Adding some error types to `Codec`, we can let all `StorageSerde` types also implement `Codec` so that they can be used seamlessly w/o the `StorageSerdeValue` newtype.

Reply:
Should we have a separate trait for key codec since that needs to be pretty much always hand crafted? (basically, renaming Codec to KeyCodec?)

Reply:
Yes, that's a good idea.